### PR TITLE
1.13.0 rc5 cherrypicks

### PR DESCRIPTION
**Pull request by vieux**
Fix inspect object by invalid reference #29681
fix swagger description for `DELETE /plugin/{name}` #29707
update experimental/README.md #28160
Add missing "--default-runtime" flag in manpage #29500
Remove bash completion for `docker node ps --all|-a` #29509
Remove `--all|-a` from `docker node ps` reference #29510
Remove bash completion for deprecated `docker daemon` #29551
Add 'volume prune' to the volume commands index #29682
Add bash completion for `plugin install --alias` #29690
Update 'ID' field for 'docker plugin ls' #29701
Fix usage of boltdb in volume restore #29664
commit: do not change container labels on commit #29550
Fix bash completion for `plugin enable|disable` #29805 
Fix image's `CMD` after `WORKDIR` in Dockerfile #29679
fix powershell dco check #29494
Trim quotes from tls flag values #29862
Fix usage message of `plugin inspect` #29806
<del>Move package cliconfig to cli/config #29313</del> (@vdemeester removed, not required)
Fixes a race condition in client events monitoring #29456
Remove `docker stack ps -a` to match removal of `docker service/node ps -a` #29716
Optimization a error description #29789
Add `--file` flag for `docker secret create` command #28629
Fix race/deadlock in v1 plugin handlers #29733
replace no-remove by sample-volume-plugin in docs #29431
Fix docker update clear restart policy of monitor #29441

### CODE REVIEW

**Comment by vieux** [comment]
ping @docker/core-engine-maintainers

**Comment by coolljt0725** [comment]
@vieux  add https://github.com/docker/docker/pull/29441 and https://github.com/docker/docker/pull/29721 ?

**Comment by vieux** [comment]
@coolljt0725  #29721  doesn't cherry-pick cleanly, can you open a PR against 1.13.x with the same changes ?

**Comment by vdemeester** [comment]
Removed #29313 and fixed compilation/test problems in the respective commits 👼

File: docs/reference/commandline/dockerd.md [hunk 1]
```diff
@@ -684,16 +684,18 @@ configuration file or using the `--add-runtime` command line argument.
...
-	"runc": {
-		"path": "runc"
-	},
-	"custom": {
-		"path": "/usr/local/bin/my-runc-replacement",
-		"runtimeArgs": [
-			"--debug"
-		]
+{
+	"default-runtime": "runc",
```
Comment (mstanleyjones) [inline docs/reference/commandline/dockerd.md]: @johndmulhausen do you need to add anything to the dockerd YAML file you made for 1.13 related to this?

File: docs/reference/commandline/events.md [hunk 2]
```diff
@@ -194,8 +194,8 @@ relative to the current time on the client machine:
     2015-12-23T21:38:25.119625123Z network connect 8b111217944ba0ba844a65b13efcd57dc494932ee2527577758f939315ba2c5b (name=test-event-network-local, container=b4be644031a3d90b400f88ab3d4bdf4dc23adb250e696b6328b85441abe2c54e, type=bridge)
 
     $ docker events --filter 'type=plugin' (experimental)
-    2016-07-25T17:30:14.825557616Z plugin pull ec7b87f2ce84330fe076e666f17dfc049d2d7ae0b8190763de94e1f2d105993f (name=tiborvass/no-remove:latest)
-    2016-07-25T17:30:14.888127370Z plugin enable ec7b87f2ce84330fe076e666f17dfc049d2d7ae0b8190763de94e1f2d105993f (name=tiborvass/no-remove:latest)
+    2016-07-25T17:30:14.825557616Z plugin pull ec7b87f2ce84330fe076e666f17dfc049d2d7ae0b8190763de94e1f2d105993f (name=tiborvass/sample-volume-plugin:latest)
```
Comment (mstanleyjones) [inline docs/reference/commandline/events.md]: Remember for 1.13 the docs no longer consume these Markdown files directly but consume YAML files (which read the cobra specifications)  to create the CLI reference.

File: docs/reference/commandline/secret_create.md [hunk 3]
```diff
@@ -16,23 +16,36 @@ keywords: ["secret, create"]
...
+Creates a secret using standard input or from a file for the secret content. You must run this
 command on a manager node.
 
 ## Examples
 
 ### Create a secret
 
 ```bash
-$ cat secret.json | docker secret create secret.json
+$ cat secret.json | docker secret create -f - secret.json
```
Comment (mstanleyjones) [inline docs/reference/commandline/secret_create.md]: Why would you do it this way?? Just `docker secret create -f secret.json secret.json

Also these examples are made more confusing when you call the secret the same as the file name used to create it. So maybe: `docker secret create -f secret.json my_secret`

File: docs/reference/commandline/secret_create.md [hunk 4]
```diff
@@ -16,23 +16,36 @@ keywords: ["secret, create"]
...
+
+$ docker secret ls
+ID                          NAME                    CREATED                                   UPDATED                                   SIZE
+mhv17xfe3gh6xc4rij5orpfds   secret.json             2016-10-27 23:25:43.909181089 +0000 UTC   2016-10-27 23:25:43.909181089 +0000 UTC   1679
+```
+
+### Create a secret with a file
+
+```bash
+$ docker secret create --file secret.in secret.json
```
Comment (mstanleyjones) [inline docs/reference/commandline/secret_create.md]: I actually think the file name should be `secret.json` and the secret name should be something else.

File: docs/reference/commandline/secret_create.md [hunk 5]
```diff
@@ -43,7 +56,7 @@ mhv17xfe3gh6xc4rij5orpfds   secret.json             2016-10-27 23:25:43.90918108
 ### Create a secret with labels
 
 ```bash
-$ cat secret.json | docker secret create secret.json --label env=dev --label rev=20161102
+$ cat secret.json | docker secret create secret.json -f - --label env=dev --label rev=20161102
```
Comment (mstanleyjones) [inline docs/reference/commandline/secret_create.md]: Again, don't cat the file if you have the `-f` flag available.

**Review by mdlinville — CHANGES_REQUESTED** [review]

**Comment by tiborvass** [comment]
LGTM

**Review by LK4D4 — APPROVED** [review]
LGTM

**Comment by tiborvass** [comment]
@mstanleyjones thanks for your input, those will have to be addressed in master first.

**Comment by mstanleyjones** [comment]
Understood and it was already done in #29889. Thanks!

File: docs/reference/commandline/dockerd.md (hunk 1 above)
Comment (johndmulhausen) [inline docs/reference/commandline/dockerd.md]: Most likely; it is [here](https://github.com/docker/docker.github.io/tree/vnext-engine/_data/dockerd-cli) and should probably live over here in docker/docker and get imported via Dockerfile or somesuch

Comment (johndmulhausen) [inline docs/reference/commandline/dockerd.md]: Filed issue https://github.com/docker/docker.github.io/issues/1038

**Pull request merged**